backend/windmill-worker/src/rust_executor.rs

```diff
@@ -23,14 +23,30 @@ use crate::{
     RUST_CACHE_DIR, TZ_ENV,
 };
 
+#[cfg(windows)]
+use crate::SYSTEM_ROOT;
+
 const NSJAIL_CONFIG_RUN_RUST_CONTENT: &str = include_str!("../nsjail/run.rust.config.proto");
 
+#[cfg(unix)]
 lazy_static::lazy_static! {
     static ref CARGO_HOME: String = std::env::var("CARGO_HOME").unwrap_or_else(|_| "/usr/local/cargo".to_string());
     static ref RUSTUP_HOME: String = std::env::var("RUSTUP_HOME").unwrap_or_else(|_| "/usr/local/rustup".to_string());
     static ref CARGO_PATH: String = format!("{}/bin/cargo", std::env::var("CARGO_HOME").unwrap_or("/usr/local/cargo/bin/cargo".to_string()));
 }
 
+#[cfg(windows)]
+lazy_static::lazy_static! {
+    static ref HOME_DIR: String = std::env::var("HOME").expect("Could not find the HOME environment variable");
+    static ref CARGO_HOME: String = std::env::var("CARGO_HOME").unwrap_or_else(|_| {
+        format!("{}\\.cargo", *HOME_DIR)
+    });
+    static ref RUSTUP_HOME: String = std::env::var("RUSTUP_HOME").unwrap_or_else(|_| {
+        format!("{}\\.rustup", *HOME_DIR)
+    });
+    static ref CARGO_PATH: String = format!("{}/bin/cargo.exe", *CARGO_HOME);
+}
```

Comment:
if only the default change then assign the unwrap_or_else to a const  default and make that be different
=>

can be factorized